arts-main/pubsub/handlers.py

```diff
@@ -81,24 +83,34 @@ def __get_runtime_or_schedule(self, msg, module):
         else:
             return self.scheduler.schedule_new_module(module)
 
-    def __create_module(self, msg):
+    def __create_module(self, msg, send_wasm=False):
         """Handle create message."""
-        data = msg.get('data')
-        if 'apis' not in data:
-            if data.get("filetype") == FileType.PY:
-                data['apis'] = ["python:python3"]
-            else:
-                data['apis'] = ["wasi:snapshot_preview1"]
-        if 'filetype' not in data:
-            if "python" in str(data.get("filename")):
-                data['filetype'] = FileType.PY
-            else:
-                data['filetype'] = FileType.WA
-
-        module = self.__object_from_dict(Module, data)
-        module.source = self.__create_or_get_file(msg)
-        module.parent = self.__get_runtime_or_schedule(msg, module)
-        module.save()
+        if(send_wasm):
+            """ If here, send the WASM file over to the runtime. """
+            module = self._get_object(
+                msg.topic, msg.get('data', 'details', 'uuid'), model=Module)
+            if(module.filetype != FileType.WA):
+                raise messages.FileNotFound(module.filename)
+            module.wasm = file_handler.get_wasm(module.filename)
+            if(module.wasm is None):
```

Comment:
Instead of returning None on FileNotFound then checking for None again, file_handler should just raise messages.FileNotFound.

Reply:
Fixed